### ISSUE TITLE
RCS1163 incorrectly reported for IEnumerables of a deconstructable type

### ISSUE THREAD

**Issue by jdh28** [JosefPihrt/Roslynator#220]
The unused parameter rule, RCS1163, is incorrectly reported when a parameter is an IEnumerable of tuples (or a type that can be deconstructed).

```c#
void EnumerateTupleList(IEnumerable<(int, string)> sequence)
{
    foreach (var (i, s) in sequence)
    {
    }
}
```

**Comment by JosefPihrt** [comment]
It seems like a bug in Roslyn. When you try to find all references for parameter 'sequence' it won't find 'sequence' in foreach.

I filed the issue in roslyn repo.

https://github.com/dotnet/roslyn/issues/23652
